packages/core/src/contrib/Constraints.ts

```diff
@@ -683,6 +684,28 @@ export const constrDict = {
     // if x < y then 0 else (x - y)^2
     return ifCond(lt(x, y), constOf(0), squared(sub(x, y)));
   },
+
+  collinear: (
+    [, p0]: [string, any],
+    [, p1]: [string, any],
+    [, p2]: [string, any]
+  ) => {
+    if (every([p0, p1, p2].map((props) => props["center"]))) {
+      const c1 = fns.center(p0);
+      const c2 = fns.center(p1);
+      const c3 = fns.center(p2);
+      const v1 = ops.vsub(c1, c2);
+      const v2 = ops.vsub(c2, c3);
+      const v3 = ops.vsub(c1, c3);
+
```

Comment:
can you add an inline comment about the math of this at a high level? it's not intuitive to me why this should work to make them collinear if it's operating purely on the norms of the vectors, not their angles